### PR TITLE
Correct container runtime status code propagation

### PR DESCRIPTION
**Pull request by onalante-msft**
During the `tokio 0.1` to `1.0` transition, non-successful error codes originating from the underlying container runtime were subsumed into `HTTP 500` responses. So, for example, it would not be possible to determine through the management API whether a module operation failed due to the module being missing or due to some other internal error. This PR restores error code propagation from the container runtime through the management API. Propagation was manually verified with `curl` in the interest of time.

The large line removal count is due to the deletion of unused code: some already unreferenced (e.g. `edgelet_core::logs`), others no longer supported (e.g. `edgelet_docker::notary`). The most substantial structural code change is in `docker::apis::client`, which now uses `hyper` directly instead of using `http_common`.

## Azure IoT Edge PR checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines and Best Practices
- [x] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request includes a concise summary of the enhancement or bug fix.

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.
- Description of the pull request includes 
	- [ ] concise summary of tests added/modified
	- [x] local testing done.

### CODE REVIEW

**Review by arsing — APPROVED** [review]

**Comment by onalante-msft** [comment]
I am not sure how a one-line change dismissed the review...

**Review by arsing — APPROVED** [review]

**Review by gordonwang0 — APPROVED** [review]
Please update the PR description before merging.

**Pull request merged**
